src/test/java/io/cdap/plugin/servicenow/source/ServiceNowMultiSourceTest.java

```diff
@@ -16,147 +16,212 @@
-
-    Mockito.doNothing().when(config).validate(mockFailureCollector);
-    OAuthClient client = Mockito.spy(new OAuthClient(new URLConnectionClient()));
-    OAuthClientRequest request = OAuthClientRequest.tokenLocation(REST_API_ENDPOINT)
-            .setGrantType(GrantType.PASSWORD)
-            .setClientId(CLIENT_ID)
-            .setClientSecret(CLIENT_SECRET)
-            .setUsername(USER)
-            .setPassword(PASSWORD)
-            .buildBodyMessage();
-
-    OAuthJSONAccessTokenResponse accessTokenResponse = new OAuthJSONAccessTokenResponse();
-    Mockito.when(client.accessToken(request, OAuth.HttpMethod.POST, OAuthJSONAccessTokenResponse.class)).
+    ServiceNowTableAPIClientImpl restApi = Mockito.mock(ServiceNowTableAPIClientImpl.class);
+    PowerMockito.whenNew(ServiceNowTableAPIClientImpl.class).
+      withArguments(Mockito.any(ServiceNowBaseSourceConfig.class)).thenReturn(restApi);
+    List<Map<String, Object>> result = new ArrayList<>();
+    Map<String, Object> map = new HashMap<>();
+    map.put("key", "value");
+    result.add(map);
+    int httpStatus = HttpStatus.SC_OK;
+    Map<String, String> headers = new HashMap<>();
+    String responseBody = "{\n" +
+      "    \"result\": []\n" +
+      "}";
+    RestAPIResponse restAPIResponse = new RestAPIResponse(httpStatus, headers, responseBody);
+    Mockito.when(restApi.executeGet(Mockito.any())).thenReturn(restAPIResponse);
+    Mockito.when(restApi.parseResponseToResultListOfMap(restAPIResponse.getResponseBody())).thenReturn(result);
+    OAuthClient oAuthClient = Mockito.mock(OAuthClient.class);
+    PowerMockito.whenNew(OAuthClient.class).
+      withArguments(Mockito.any(URLConnectionClient.class)).thenReturn(oAuthClient);
+    OAuthJSONAccessTokenResponse accessTokenResponse = Mockito.mock(OAuthJSONAccessTokenResponse.class);
+    Mockito.when(oAuthClient.accessToken(Mockito.any(), Mockito.anyString(), Mockito.any(Class.class))).
       thenReturn(accessTokenResponse);
     Mockito.when(accessTokenResponse.getAccessToken()).thenReturn("token");
-
+    RestAPIResponse response = Mockito.spy(restAPIResponse);
+    CloseableHttpClient httpClient = Mockito.mock(CloseableHttpClient.class);
+    HttpClientBuilder httpClientBuilder = Mockito.mock(HttpClientBuilder.class);
+    PowerMockito.mockStatic(HttpClientBuilder.class);
+    PowerMockito.mockStatic(RestAPIResponse.class);
+    PowerMockito.when(HttpClientBuilder.create()).thenReturn(httpClientBuilder);
+    Mockito.when(httpClientBuilder.build()).thenReturn(httpClient);
+    CloseableHttpResponse httpResponse = Mockito.mock(CloseableHttpResponse.class);
+    Mockito.when(httpClient.execute(Mockito.any())).thenReturn(httpResponse);
+    PowerMockito.when(RestAPIResponse.parse(ArgumentMatchers.any(), ArgumentMatchers.anyString())).
+      thenReturn(response);
     serviceNowMultiSource.prepareRun(context);
     Assert.assertEquals(0, mockFailureCollector.getValidationFailures().size());
```

Comment:
From past google reviews I have seen that it's better to provide 3 parameter in assertEquals, "Failure msg", "expected value", "actual value"

Reply:
Added the failure message. c0e6a0f